components/formats-bsd/src/loci/formats/in/MicromanagerReader.java

```diff
@@ -711,8 +710,11 @@ else if (s.length() > 100 * 1024 * 1024) {
     }
 
     byte[] b = new byte[(int) s.length()];
-    s.readFully(b);
-    s.close();
+    try {
```

Comment:
I assume the `RandomAccessInputStream s = new RandomAccessInputStream(jsonData);` line above could be wrapped in a more generic `try-with-resources` like other changs in the code

Reply:
This is fixed in another PR https://github.com/openmicroscopy/bioformats/pull/3277
I was trying to split the work but maybe it is too confusing

Reply:
I managed to create conflict during the cleaning
So probably better to combine #3277 and this one